azkaban-common/src/main/java/azkaban/utils/Props.java

```diff
@@ -811,6 +812,30 @@ public void storeFlattened(OutputStream out) throws IOException {
   }
 
   /**
+   * Returns a map of all the flattened properties
+   *
+   * @Return
+   */
+  public Map<String,String> getFlattened(){
+
+    // creating a treeMap here so that the flattened list can be sorted
+    // alphabetically by the key value.
+    Map<String,String> returnVal = new TreeMap<String,String>();
+
+    // to keep the logic in sync with the rest piece of code,
+    // when there is a conflict, value from the child takes the priority.
+    for (Props curr = this; curr != null; curr = curr.getParent()) {
+      for (String key : curr.localKeySet()) {
+          if (!returnVal.containsKey(key)){
```

Comment:
I think it is, once the iterator moves to the parent (which means the curr is now pointing to the parent) anything that previously has been defined by the child shouldn't be overwritten, the check in place is to make sure the rule is followed. Please let me know if I understand the concern correctly.